Quick enter button

loic-martinez-freelance opened the pull request:
Added a quick enter button on the giveaway list. (With a setting to
enable / disable it)

Commit:
Quick enter button

Added a quick enter button on the giveaway list. (With a setting to
enable / disable it)

Commit:
Merge remote-tracking branch 'original/master'

loic-martinez-freelance commented:
Added a screenshot

<img src="https://cloud.githubusercontent.com/assets/1225847/21809363/a87c1bcc-d747-11e6-8357-bd35025f5056.png" width="40%">

mabako commented on core/src/main/java/net/mabako/steamgifts/adapters/viewholder/GiveawayListItemViewHolder.java:
```diff
@@ -136,6 +140,35 @@ public void onError() {
+            } else {
+                giveawayEnterButton.setText(activity.getString(R.string.enter_giveaway));
+                //Check if the user have enough points, that the giveaway is not from him and that he have the required level
+                giveawayEnterButton.setEnabled(giveaway.getPoints() <= SteamGiftsUserData.getCurrent(null).getPoints() && giveaway.getLevel() <= SteamGiftsUserData.getCurrent(null).getLevel() && !SteamGiftsUserData.getCurrent(null).getName().equals(giveaway.getCreator()));
```
How does the context menu determine that you can enter a giveaway? This looks like it should be a shared function.

mabako commented on core/src/main/java/net/mabako/steamgifts/adapters/viewholder/GiveawayListItemViewHolder.java:
```diff
@@ -136,6 +140,35 @@ public void onError() {
+        } else {
+            // Set the correct text based on the giveaway is already entered or not
+            if (giveaway.isEntered()) {
+                giveawayEnterButton.setText(activity.getString(R.string.leave_giveaway));
```
I would prefer if this was an icon from font-awesome. The list of hidden games is an example where a "x" is used for the quick action, I think, which I feel is more appropriate. 

This issue likely is worse on low resolution devices.

mabako commented on core/src/main/res/values/strings.xml:
```diff
@@ -256,4 +256,7 @@
     <!-- unsorted -->
     <string name="could_not_load_image">Could not load image.</string>
     <string name="share_image">Share Image</string>
+    <string name="preference_giveaway_quick_enter">Display quick enter button</string>
```
Note to self: check for consistency when I'm home

mabako commented on core/src/main/java/net/mabako/steamgifts/adapters/viewholder/GiveawayListItemViewHolder.java:
```diff
@@ -136,6 +140,35 @@ public void onError() {
+        // Initialize the enter button
+        // Check if logged or the quick enter button setting is enabled
+        boolean loggedIn = SteamGiftsUserData.getCurrent(null).isLoggedIn();
+        if (!loggedIn || !PreferenceManager.getDefaultSharedPreferences(fragment.getContext()).getBoolean("preference_giveaway_show_quick_enter", false)){
```
The default value in your settings definition is true, but is assumed to be false here.

mabako commented on core/src/main/java/net/mabako/steamgifts/fragments/GiveawayListFragment.java:
```diff
@@ -200,7 +200,8 @@ public void onEnterLeaveResult(String giveawayId, String what, Boolean success,
                 } else {
-                    adapter.notifyItemChanged(giveaway);
+                    // We refresh the entire dataset to let the other quick enter buttons know if the user still have enough points
+                    adapter.notifyDataSetChanged();
```
What are the performance implications for this?  Would it make sense to only refresh all items if the setting is enabled?

mabako reviewed (commented):
I'll have a more in-depth look at it when I'm back home (probably tomorrow), but apart from disliking the button appearance it looks good.

Commit:
Corrected the default value for the quick enter button
There is now a function in the Giveaway class to know if the current user can enter the giveaway

mabako committed:
fixed layout params

mabako committed:
minor refactoring and use fontawesome-icons

mabako committed:
settings are now a bit leaner

mabako commented:
Thanks for your pull request!
